rust/eos-rs/src/api/dungeon_mode/dungeon_struct.rs

```diff
@@ -0,0 +1,1143 @@
+    /// (it's usually set  to 1 or 0). If the weather's source is removed, this value becomes the
+    /// normal number of turns  left for that weather condition. Priority in the event of multiple
+    /// nonzero counters is the same as with weather_turns.
+    pub fn get_artificial_permaweather_turns(&self) -> &[u16; 8] {
+        &self.0.as_ref().artificial_permaweather_turns
+    }
+
+    /// Turns left for artificial permaweather from weather-setting abilities like Drought,
+    /// Sand Stream, Drizzle, and Snow Warning; one counter for each weather type in enum
+    /// weather_id (except WEATHER_RANDOM). Any nonzero value triggers that weather condition
+    /// (it's usually set  to 1 or 0). If the weather's source is removed, this value becomes the
+    /// normal number of turns  left for that weather condition. Priority in the event of multiple
+    /// nonzero counters is the same as with weather_turns.
+    pub fn set_artificial_permaweather_turns(&mut self) -> &mut [u16; 8] {
+        &mut self.0.as_mut().artificial_permaweather_turns
+    }
+
+    /// For damaging weather conditions like sandstorm. Counts down from 9-0, damage on 9.
+    pub fn get_weather_damage_counter(&self) -> u8 {
+        self.0.as_ref().weather_damage_counter
+    }
+
+    /// Sets the weather damage counter.
+    pub fn set_weather_damage_counter(&mut self, counter: u8) {
+        self.0.as_mut().weather_damage_counter = counter
+    }
+
+    /// Number of turns left for the Mud Sport condition.
+    pub fn get_mud_sport_turns(&self) -> u8 {
+        self.0.as_ref().mud_sport_turns
+    }
+
+    /// Sets the number of turns left for the Mud Sport condition.
+    pub fn set_mud_sport_turns(&mut self, counter: u8) {
+        self.0.as_mut().mud_sport_turns = counter
+    }
+
+    /// Number of turns left for the Water Sport condition.
+    pub fn get_water_sport_turns(&self) -> u8 {
+        self.0.as_ref().water_sport_turns
+    }
+
+    /// Sets the number of turns left for the Water Sport condition.
+    pub fn set_water_sport_turns(&mut self, counter: u8) {
+        self.0.as_mut().water_sport_turns = counter
+    }
+
+    /// Whether or not current weather is nullified by Cloud Nine or Air Lock.
+    pub fn is_weather_nullified(&self) -> bool {
+        self.0.as_ref().nullify_weather > 0
+    }
+
+    /// Sets whether or not current weather is nullified by Cloud Nine or Air Lock.
+    pub fn set_weather_nullified(&mut self, nullified: bool) {
+        self.0.as_mut().nullify_weather = nullified as ffi::bool_
+    }
+
+    /// Whether or not Gravity is in effect.
+    pub fn is_gravity_in_effect(&self) -> bool {
+        self.0.as_ref().gravity > 0
+    }
+
+    /// Sets whether or not Gravity is in effect.
+    pub fn set_gravity_in_effect(&mut self, gravity: bool) {
+        self.0.as_mut().gravity = gravity as ffi::bool_
+    }
+
+    /// Entity that is taking it's turn at this moment.
```

Comment:
its

Reply:
:pensive:

Not at my PC anymore but pretty sure I copied that from the C headers? Is it wrong there too?

Reply:
>Is it wrong there too?

No, actually. In the headers it's its but it's it's here 😛.

Reply:
:realshaymin: